### PR TITLE
feat(terraform): add CKV NCP rules about NKSPublicAccess

### PR DESCRIPTION
**Pull request by pj991207**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description
Add terraform scan rules about provider Naver Cloud Platform.
CKV_NCP_19 is associated with NKSPublicAccess

### Description
https://registry.terraform.io/providers/NaverCloudPlatform/ncloud/latest/docs/resources/nks_cluster

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by nimrodkor — APPROVED** [review]

File: tests/terraform/checks/resource/ncp/example_NksPublicAccess/main.tf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  kube_network_plugin         = "cilium"
+  subnet_no_list              = [ ncloud_subnet.subnet.id ]
+  vpc_no                      = ncloud_vpc.vpc.id
+  zone                        = "KR-1"
+  public_network = false
+  log {
+    audit = true
+  }
+}
+resource "ncloud_nks_cluster" "fail" {
```
Comment (gruebel) [inline tests/terraform/checks/resource/ncp/example_NksPublicAccess/main.tf:16]: can you also add an example without setting `public_network` then it is public or automatically private?

Comment (pj991207) [reply tests/terraform/checks/resource/ncp/example_NksPublicAccess/main.tf]: I fixed my code about your advice. when I checked default options about public_network, It was 'False' so, I reflected it.

File: checkov/terraform/checks/resource/ncp/NKSPublicAccess.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+
+
+class NKSPublicAccess(BaseResourceCheck):
```
Comment (gruebel) [inline checkov/terraform/checks/resource/ncp/NKSPublicAccess.py:5]: you actually want to use negative resource value check here. Here is an example of how it looks like https://github.com/bridgecrewio/checkov/blob/master/checkov/terraform/checks/resource/aws/DMSReplicationInstancePubliclyAccessible.py

**Review by gruebel — APPROVED** [review]
great work 💯

**Pull request merged**